projects/ad9545/src/examples/basic/basic_example.c

```diff
@@ -0,0 +1,275 @@
+	ad9545_ip.ref_in_clks[2].d_tol_ppb = 10000000;
+	ad9545_ip.ref_in_clks[2].monitor_hyst_scale = 7; // 87500
+	ad9545_ip.ref_in_clks[2].valid_t_ms = 10;
+	ad9545_ip.ref_in_clks[2].freq_thresh_ps = 2000;
+	ad9545_ip.ref_in_clks[2].phase_thresh_ps = 2000;
+	ad9545_ip.ref_in_clks[2].freq_lock_fill_rate = 20;
+	ad9545_ip.ref_in_clks[2].freq_lock_drain_rate = 20;
+	ad9545_ip.ref_in_clks[2].phase_lock_fill_rate = 20;
+	ad9545_ip.ref_in_clks[2].phase_lock_drain_rate = 20;
+	/* REF-BB */
+	ad9545_ip.ref_in_clks[3].ref_used = true;
+	ad9545_ip.ref_in_clks[3].address = 3;
+	ad9545_ip.ref_in_clks[3].mode = AD9545_SINGLE_ENDED;
+	ad9545_ip.ref_in_clks[3].s_conf = DRIVER_MODE_DC_COUPLED_1V8;
+	ad9545_ip.ref_in_clks[3].r_div_ratio = 1;
+	ad9545_ip.ref_in_clks[3].d_tol_ppb = 100000;
+	ad9545_ip.ref_in_clks[3].monitor_hyst_scale = 3; // 12500
+	ad9545_ip.ref_in_clks[3].valid_t_ms = 1000;
+	ad9545_ip.ref_in_clks[3].freq_thresh_ps = 2000;
+	ad9545_ip.ref_in_clks[3].phase_thresh_ps = 2000;
+	ad9545_ip.ref_in_clks[3].freq_lock_fill_rate = 100;
+	ad9545_ip.ref_in_clks[3].freq_lock_drain_rate = 10;
+	ad9545_ip.ref_in_clks[3].phase_lock_fill_rate = 100;
+	ad9545_ip.ref_in_clks[3].phase_lock_drain_rate = 10;
+
+
+	/* Initialize Fixed clocks to use as input references */
+	ret = no_os_clk_init(&ref_b_clk, &refb_clk_ip);
+	if (ret)
+		return ret;
+
+	ret = no_os_clk_init(&ref_bb_clk, &refbb_clk_ip);
+	if (ret)
+		return ret;
+
+	ret = no_os_clk_init(&ref_m1_clk, &refm1_clk_ip);
+	if (ret)
+		return ret;
+
+	ad9545_ip.ref_in_clks[3].parent_clk = ref_bb_clk;
+	ad9545_ip.ref_in_clks[2].parent_clk = ref_b_clk;
+	ad9545_ip.aux_tdc_clks[AD9545_CLK_AUX_TDC0].parent_clk = ref_m1_clk;
+
+	printf("AD9545 basic_example\n\r");
+
+	ret = ad9545_init(&ad9545_dev, &ad9545_ip);
+	if (ret < 0)
+		return ret;
+
+	ret = ad9545_setup(ad9545_dev);
+	if (ret < 0)
+		goto error;
+
+	/* Set Clocks' rate */
+	ret = no_os_clk_set_rate(ad9545_dev->clks[AD9545_CLK_NCO][AD9545_NCO0],
+				 10000);
+	if (ret)
+		goto error;
+
+	ret = no_os_clk_set_rate(ad9545_dev->clks[AD9545_CLK_PLL][AD9545_PLL0],
+				 1400000000);
+	if (ret)
+		goto error;
+
+	ret = no_os_clk_set_rate(ad9545_dev->clks[AD9545_CLK_PLL][AD9545_PLL1],
+				 1750000000);
+	if (ret)
+		goto error;
+
+	ret = no_os_clk_set_rate(ad9545_dev->clks[AD9545_CLK_OUT][AD9545_Q0A],
+				 10000000);
+	if (ret)
+		goto error;
+
+	ret = no_os_clk_set_rate(ad9545_dev->clks[AD9545_CLK_OUT][AD9545_Q0B],
+				 10000000);
+	if (ret)
+		goto error;
+
+	ret = no_os_clk_set_rate(ad9545_dev->clks[AD9545_CLK_OUT][AD9545_Q1A],
+				 25000000);
+	if (ret)
+		goto error;
+
+	ret = no_os_clk_set_rate(ad9545_dev->clks[AD9545_CLK_OUT][AD9545_Q1B],
+				 10000000);
+	if (ret)
+		goto error;
+
+	ret = no_os_clk_set_rate(
+		      ad9545_dev->clks[AD9545_CLK_AUX_TDC][AD9545_CLK_AUX_TDC0],
+		      200000);
+	if (ret)
+		goto error;
+
+	/* Recalibrate PLLs */
+	ret = ad9545_calib_aplls(ad9545_dev);
+	if (ret < 0)
+		return ret;
+
+	while(1) {
```

Comment:
the program could finish with a "Done" message if everything was ok. Do we need the loop here after the device is setup?

Reply:
Yes, indeed the loop is not necessary, maybe i left there after the tests, i will remove.